### ISSUE TITLE
introduce torch.tpu.is_available()

### ISSUE THREAD

**Issue by miladm** [pytorch/xla#7767]
## 🐛 Usability Bug

If a user wants to check for a `cuda` device, they call: `torch.cuda.is_available()`

In torch_xla, similar information can be extracted in a few ways. This simple API should not be defined by the user. 

Let's enumerate a few approaches and think of a more native alternative for users to experience consistency.

Current:
```
def is_tpu_available():
    # Return if the device is TPU
    devices = xm.xla_real_devices()
    if len(devices) > 0:
        return 'TPU' in devices[0]
    return False
```
Suggestion: `torch.tpu.is_available()`

Current:
```
def xla_device_type():
    # Return "TPU", "CPU", "GPU"
    device = xm.xla_device()
    return xm.xla_device_hw(device)
```
Suggestion: `torch.xla.device_type()`

Current:
```
def is_xla_available():
    # Return if the device is XLA
    return `xla` in str(xm.xla_device())
```
Suggestion: ```torch.xla.is_available()```

**Comment by will-cromar** [comment]
Is this a duplicate of #7765?

**Comment by miladm** [comment]
no, this issue is about having an `is_available()` API. The other issue is related to `xmp.spawn()` behavior.

**Comment by miladm** [comment]
cc @yaoshiang - can we discuss this in our future conversations?

cc @pgmoka
